code/Helper/Image.php

```diff
@@ -31,16 +31,11 @@ public function toString()
             $model->setWatermark($this->getWatermark());
         }
 
-        return $this->getUrl($model->saveFile());
+        return $this->removeProtocol($model->saveFile()->getUrl());
     }
 
-    public function getUrl(Mage_Catalog_Model_Product_Image $model)
+    public function removeProtocol($url)
     {
-        $baseDir = Mage::getBaseDir();
-        $path = str_replace($baseDir.DS, '', $model->getNewFile());
-
-        $url = str_replace(DS, '/', $path);
-
-        return $url;
+        return str_replace(['https://', 'http://'], '//', $url);
```

Comment:
I solved it (at work, that why I couldn't share it yet) by removing 'media' from $url and then passing the configuration media url along with the hit template.

Like this:  $path = str_replace($baseDir . DS . 'media' . DS, '', $model->getNewFile());

That way you don't have to store the same media url on each record for products and categories.